### PR TITLE
Failover priority configuration elements

### PR DESCRIPTION
**Pull request by albinsuresh**

### CODE REVIEW

File: tcconfig-schema/src/main/xsd/terracotta.xsd [hunk 1]
```diff
@@ -166,7 +178,24 @@
     <xs:attribute name="name" type="xs:string" use="required"/>
     <xs:attribute name="value" type="xs:string" use="required"/>
   </xs:complexType>
-  
+
+  <xs:complexType name="failover-priority">
+    <xs:choice>
+      <xs:element name="consistency" type="tc:consistency" minOccurs="0"/>
```
Comment (ljacomet) [inline tcconfig-schema/src/main/xsd/terracotta.xsd]: What's the point of `minOccurs` here?

File: tcconfig-schema/src/main/xsd/terracotta.xsd [hunk 2]
```diff
@@ -166,7 +178,24 @@
...
+      <xs:element name="availability" type="xs:string" fixed=""/>
```
Comment (ljacomet) [inline tcconfig-schema/src/main/xsd/terracotta.xsd]: can you explain what `fixed` means?

File: tcconfig-schema/src/main/xsd/terracotta.xsd [hunk 3]
```diff
@@ -166,7 +178,24 @@
...
+    </xs:choice>
+  </xs:complexType>
+
+  <xs:complexType name="consistency">
+    <xs:all minOccurs="0">
+      <xs:element name="voter" type="tc:voter" minOccurs="0"/>
```
Comment (ljacomet) [inline tcconfig-schema/src/main/xsd/terracotta.xsd]: There seems to be too many `minOccurs="0"` here

File: tcconfig-schema/src/main/xsd/terracotta.xsd (hunk 2 above)
Comment (albinsuresh) [reply tcconfig-schema/src/main/xsd/terracotta.xsd]: This is just a trick that I used define this as an empty element that does not take any attributes or sub-elements.

File: tcconfig-schema/src/main/xsd/terracotta.xsd (hunk 3 above)
Comment (albinsuresh) [reply tcconfig-schema/src/main/xsd/terracotta.xsd]: Voter tags are really optional. You can define the consistency without defining any voter tags and that's still valid.

File: tcconfig-schema/src/main/xsd/terracotta.xsd (hunk 1 above)
Comment (albinsuresh) [reply tcconfig-schema/src/main/xsd/terracotta.xsd]: That one was unintentional. Will remove.

Comment (albinsuresh) [reply tcconfig-schema/src/main/xsd/terracotta.xsd]: That one was unintentional. Will remove.

**Pull request merged**